### PR TITLE
improve typings of neotest adapter interface

### PR DESCRIPTION
**Pull request by bram209**
by using https://github.com/folke/neodev.nvim as per suggestion on the `neotest` repo readme

### CODE REVIEW

File: lua/neotest-rust/init.lua [hunk 1]
```diff
@@ -31,6 +31,11 @@ local cargo_metadata = setmetatable({}, {
     end,
 })
 
+---Find the project root directory given a current directory to work from.
+---Should no root be found, the adapter can still be used in a non-project context if a test file matches.
+---@async
```
Comment (rouge8) [inline lua/neotest-rust/init.lua:36]: what makes this async? `cargo_metadata()`?

**Review by rouge8 — COMMENTED** [review]
are there any tools to check these type hints?

File: lua/neotest-rust/init.lua (hunk 1 above)
Comment (bram209) [reply lua/neotest-rust/init.lua:36]: I copied interface definitions from: https://github.com/nvim-neotest/neotest/blob/master/lua/neotest/adapters/interface.lua

Applying `@async` will give additional diagnostics: https://github.com/LuaLS/lua-language-server/wiki/Diagnostics#await

But I think in this case it is not necessary indeed

**Comment by bram209** [comment]
> are there any tools to check these type hints?

Yes, `lua_ls` can do type checking, see: https://github.com/LuaLS/lua-language-server/wiki/Type-Checking#how-it-works
`neodev.nvim` can configure `lua_ls` automatically, so that the language server will give better editor support when calling code from other libraries (neotest in this case), while also being able to resolve its type definitions (the ones here: https://github.com/nvim-neotest/neotest/blob/master/lua/neotest/types/init.lua)

**Review by rouge8 — APPROVED** [review]

**Pull request merged**